### PR TITLE
Add item durability

### PR DESCRIPTION
**Pull request by samuelWilliams99**
An item can define `hasDurability`, `durabilityUse`, `maxDurability` and `durabilityUseRandomRange`.
Adds a durability bar on item slots
Defines a `LoseDurability()` function on the weapon when it is spawned if the item it is assigned to has durability.
The SWEP is expected to call this whenever used in some way that would damage it.

### CODE REVIEW

File: gamemode/modules/inventory/sv_inventory.lua [hunk 1]
```diff
@@ -239,6 +239,25 @@ function inv.getSelectedItem( ply )
     return itemData.item, itemData.count
 end
 
+local function addDurabilityFuncs( wep, instance, itemData )
+    function wep:LoseDurability()
```
Comment (ghost) [inline gamemode/modules/inventory/sv_inventory.lua:243]: Is this function unused? It's not returned nor called so unless there is something about lua that I never knew about I feel like there is a problem here.

**Review by ghost — CHANGES_REQUESTED** [review]

File: gamemode/modules/inventory/sv_inventory.lua (hunk 1 above)
Comment (ghost) [reply gamemode/modules/inventory/sv_inventory.lua:243]: Nvm my bad this is fine

**Review by ghost — APPROVED** [review]
Looks good to me

**Review by overlordbleck — APPROVED** [review]
hot

**Pull request merged**